lib/components/base-components/PrimitiveComponent.ts

```diff
@@ -108,6 +108,8 @@ export abstract class PrimitiveComponent<
   schematic_component_id: string | null = null
   pcb_component_id: string | null = null
   cad_component_id: string | null = null
+  subcircuit_id: string | null = null
+  pcb_group_id: string | null = null
```

Comment:
I think it's better to derive these on the fly rather than set them.

Generally you only want to store the ids you're creating so that there's only one source of truth

```suggestion
```